src/api/convert_to_json.cpp

```diff
@@ -0,0 +1,116 @@
+#include <algorithm>
+#include <string>
+#include <vector>
+#include <functional>
+#include <sstream>
+#include <migraphx/errors.hpp>
+#include <migraphx/ranges.hpp>
+#include "convert_to_json.hpp"
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+using token = std::pair<const char*, const char*>;
+using lexer = std::function<const char*(const char* start, const char* end)>;
+
+template <class P>
+auto lex_while(P p)
+{
+    return [=](const char* start, const char* end) {
+        return std::find_if(start, end, [&](char c) { return not p(c); });
+    };
+}
+
+template <class P>
+auto lex_if(P p)
+{
+    return [=](const char* start, const char*) {
+        if(p(*start))
+            return start + 1;
+        return start;
+    };
+}
+
+std::vector<token> tokenize(const char* start, const char* end, std::vector<lexer> lexers)
+{
+    std::vector<token> result;
+    while(start != end)
+    {
+        bool error = true;
+        for(auto l : lexers)
+        {
+            auto next = l(start, end);
+            if(next != start)
+            {
+                if(not std::all_of(start, next, &isspace))
```

Comment:
This condition should be removed, so we dont strip whitespace in the output.

Reply:
removed.